packages/malloy/src/malloy.ts

```diff
@@ -3882,49 +3905,148 @@ export class CSVWriter extends DataWriter {
     return value;
   }
 
-  private stringify(cell: DataColumn) {
-    if (cell.isNull()) {
-      return '';
-    } else if (
-      cell.isArray() ||
-      cell.isRecord() ||
-      cell.isBoolean() ||
-      cell.isNumber()
-    ) {
-      return JSON.stringify(cell.value);
-    } else if (cell.isDate() || cell.isTimestamp()) {
-      return cell.value.toISOString();
-    } else if (cell.isString()) {
-      return cell.value;
+  // Re-using the old stringify method for sanity.
+  private stringify(value: QueryValue) {
+    if (value === null) {
+      return this.emptyCell;
+    } else if (value instanceof Date) {
+      return value.toISOString();
```

Comment:
Is it OK that this behavior has changed from converting dates and timestamps to ISO strings to just dates?

Reply:
This is the tradeoff if we want to be able to support pivoted tables in the CSVWriter. When we have pivoted data, the re-constructed result object (TODO) would not have structdef/modeldef. Hence we have to inject the data as a raw JSON object in the writer, where we lose the ability to identify timestamps 
```
export type QueryScalar = string | boolean | number | Date | Buffer | null;
```

Unless we want to have a separate CSV writer for pivoted case, which adds to the complexity in VSCode where we have to constantly look for pivoted data, this is inevitable.

Reply:
@carlineng